src/components/navBar/navBar.js

```diff
@@ -103,23 +106,68 @@ function MdNavBar($mdAria) {
       'mdSelectedNavItem': '=?',
       'navBarAriaLabel': '@?',
     },
-    template:
-      '<div class="md-nav-bar">' +
-        '<nav role="navigation">' +
-          '<ul class="_md-nav-bar-list" layout="row" ng-transclude role="listbox"' +
-            'tabindex="0"' +
-            'ng-focus="ctrl.onFocus()"' +
-            'ng-blur="ctrl.onBlur()"' +
-            'ng-keydown="ctrl.onKeydown($event)"' +
-            'aria-label="{{ctrl.navBarAriaLabel}}">' +
-          '</ul>' +
-        '</nav>' +
-        '<md-nav-ink-bar></md-nav-ink-bar>' +
-      '</div>',
-    link: function(scope, element, attrs, ctrl) {
+    template: '<div class=\'md-nav-wrapper\'> ' +
+                '<div class=\'md-nav-pagination\'> ' +
+                  '<md-prev-button ' +
+                  'aria-label="Previous Page" ' +
+                  'aria-disabled="{{!ctrl.canPageBack()}}" ' +
+                  'ng-disabled="{{!ctrl.canPageBack()}}" ' +
+                  'ng-if="ctrl.shouldPaginate" ' +
+                  'ng-click="ctrl.previousPage()"> ' +
+                    '<md-icon md-svg-src="' + $$mdSvgRegistry.mdTabsArrow + '"></md-icon> ' +
+                  '</md-prev-button> ' +
+                  '<md-next-button ' +
+                  'aria-label="Next Page" ' +
+                  'aria-disabled="{{!ctrl.canPageForward()}}" ' +
+                  'ng-disabled="{{!ctrl.canPageForward()}}" ' +
+                  'ng-if="ctrl.shouldPaginate" ' +
+                  'ng-click="ctrl.nextPage()"> ' +
+                    '<md-icon md-svg-src="' + $$mdSvgRegistry.mdTabsArrow + '"></md-icon> ' +
+                  '</md-next-button> ' +
+                '</div>' +
+                '<div class="md-nav-bar" ' +
+                'md-swipe-left="ctrl.nextPage()" ' +
+                'md-swipe-right="ctrl.previousPage()"> ' +
+                  '<nav role="navigation"> ' +
+                    '<ul class="_md-nav-bar-list" ng-transclude ' +
+                    'layout="row" ' +
+                    'role="listbox"' +
+                    'tabindex="0"' +
+                    'ng-focus="ctrl.onFocus()" ' +
+                    'ng-blur="ctrl.onBlur()" ' +
+                    'ng-keydown="ctrl.onKeydown($event)" ' +
+                    'aria-label="{{ctrl.navBarAriaLabel}}"> ' +
+                    '</ul> ' +
+                  '</nav> ' +
+                  '<md-nav-ink-bar></md-nav-ink-bar> ' +
+                '</div> ' +
+              '</div> ',
+    link: function (scope, element, attrs, ctrl) {
       if (!ctrl.navBarAriaLabel) {
         $mdAria.expectAsync(element, 'aria-label', angular.noop);
       }
+
+      var observer = new MutationObserver(function (mutations) {
+        ctrl.updatePagination();
```

Comment:
I'd say that the mutation observer is probably better than a $watch (we did a similar migration from $watch to MutationObserver for the tooltips), but it would be good if it didn't fire as often. For example, try seeing if you can get rid of any of the options in the `config`, without breaking functionality. In addition, you can debounce the callback so it doesn't fire too much if a few mutations happen together, but you might want to play around and see what works, since it might make the UI look unresponsive.